x-pack/plugins/ingest_manager/server/services/agents/checkin/state_new_actions.ts

```diff
@@ -122,10 +142,14 @@ async function createAgentActionFromPolicyAction(
   );
 
   // Mutate the policy to set the api token for this agent
-  newAgentAction.data.config.outputs.default.api_key = await getOrCreateAgentDefaultOutputAPIKey(
-    soClient,
-    agent
-  );
+  const apiKey = await getOrCreateAgentDefaultOutputAPIKey(soClient, agent);
+  if (newAgentAction.data.policy) {
+    newAgentAction.data.policy.outputs.default.api_key = apiKey;
+  }
+  // BWC for agent <= 7.9
```

Comment:
what about 7.9.1 and 7.9.2 which are already realeased?

Reply:
i see in test it will work,

Reply:
yep, <= 7.9.x